src/Mod/Draft/draftgeoutils/geometry.py

```diff
@@ -196,70 +196,57 @@ def findDistance(point, edge, strict=False):
-    startPoint = edge.valueAt(edge.FirstParameter)
-    endPoint = edge.valueAt(edge.LastParameter)
-    midParameter = (edge.FirstParameter
-                    + (edge.LastParameter - edge.FirstParameter) / 2)
-    midPoint = edge.valueAt(midParameter)
-    v1 = midPoint - startPoint
-    v2 = midPoint - endPoint
-    n = v1.cross(v2)
-    n.normalize()
-    return n
-
-
-def getNormal(shape):
+
+    if isStraightLine(shape, tol):
+        return None
+
+    plane = edge.findPlane(tol)
+    if plane:
+        normal = plane.Axis 
+        return normal
+    else:
+        return None
+
+
+def getNormal(shape, tol=-1):
     """Find the normal of a shape or list of points, if possible."""
+    
+    # for points
     if isinstance(shape, (list, tuple)):
-        if len(shape) >= 3:
-            v1 = shape[1].sub(shape[0])
-            v2 = shape[2].sub(shape[0])
-            n = v2.cross(v1)
-            if n.Length:
-                return n
-        return None
+        if len(shape) <= 2:
+            return None
+        else:
+            poly = Part.makePolygon(shape)
+            if isStraightLine(poly, tol):
+                return None
 
-    n = App.Vector(0, 0, 1)
-    if shape.isNull():
-        return n
-
-    if (shape.ShapeType == "Face") and hasattr(shape, "normalAt"):
-        n = shape.copy().normalAt(0.5, 0.5)
-    elif shape.ShapeType == "Edge":
-        if geomType(shape.Edges[0]) in ["Circle", "Ellipse"]:
-            n = shape.Edges[0].Curve.Axis
-        elif (geomType(shape.Edges[0]) == "BSplineCurve"
-              or geomType(shape.Edges[0]) == "BezierCurve"):
-            n = getSplineNormal(shape.Edges[0])
+            plane = poly.findPlane(tol)
+            if plane:
+                normal = plane.Axis
+                return normal
+            else:
+                return None
+
+    # for shapes
+    if isStraightLine(shape, tol):
+        return None
+    
     else:
-        for e in shape.Edges:
-            if geomType(e) in ["Circle", "Ellipse"]:
-                n = e.Curve.Axis
-                break
-            elif (geomType(e) == "BSplineCurve"
-                  or geomType(e) == "BezierCurve"):
-                n = getSplineNormal(e)
-                break
-
-            e1 = vec(shape.Edges[0])
-            for i in range(1, len(shape.Edges)):
-                e2 = vec(shape.Edges[i])
-                if 0.1 < abs(e1.getAngle(e2)) < 3.14:
-                    n = e1.cross(e2).normalize()
-                    break
+        plane = shape.findPlane(tol)
+        if plane:
+            normal = plane.Axis
+        else:
+            return None
 
     # Check the 3D view to flip the normal if the GUI is available
     if App.GuiUp:
-        vdir = gui_utils.get_3d_view().getViewDirection()
-        if n.getAngle(vdir) < 0.78:
-            n = n.negative()
-
-    if not n.Length:
-        return None
+        vDir = gui_utils.get_3d_view().getViewDirection()
+        if normal.getAngle(vDir) < 0.78:
```

Comment:
0.78 rad = 44.69071 degrees?

Reply:
Yes. Why do we use this number specifically?

Reply:
yes, is the value that was already in the code.